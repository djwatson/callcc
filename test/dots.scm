(import (scheme r5rs))
(define-syntax foo
  (syntax-rules ()
    ((_ (a ...) (b ...) ...) '((a  (b ...) ) ...))))

(display (foo (1 2 3) (4 4) (5 5) ))
(newline)

(define-syntax foo
  (syntax-rules ()
    ((_ (a b ...) ...) '(((a  b ) ...) ...))))

(display (foo ((1 2 3) (4 4) (5 5)) ))
(newline)

(define-syntax foo
  (syntax-rules ()
    ((_ a b ...) '((a  b ) ...))))

(display (foo (1 2 3) (4 4) (5 5) ))
(newline)

;; Too many ellipsis

;; (define-syntax foo
;;   (syntax-rules ()
;;     ((_  a b ...) '(( ((a b) ...)  ...)))))

;; (display (foo (1 2 3) (4 4 4) (5 5 5) ))

;; Too few ellipsis

;; (define-syntax foo
;;   (syntax-rules ()
;;     ((_ a ...) a)))
;; (foo 1)

