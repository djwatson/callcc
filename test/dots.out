((1 (4 4)) (2 (5 5)))
((((1 2 3) (4 4)) ((1 2 3) (5 5))))
(((1 2 3) (4 4)) ((1 2 3) (5 5)))
